src/adapter/src/catalog/apply.rs

```diff
@@ -43,49 +44,79 @@ use tracing::warn;
 
 use crate::catalog::{BuiltinTableUpdate, Catalog, CatalogState};
 
-/// Sorted [`StateUpdate`]s. Builtin view updates are stored separately, because they are loaded
+/// Sorted [`StateUpdate`]s. Builtin view additions are stored separately, because they are applied
 /// via a different mechanism than all other update types.
 struct SortedUpdates {
-    pre_builtin_view_updates: Vec<StateUpdate>,
-    builtin_view_updates: Vec<(&'static Builtin<NameReference>, GlobalId)>,
-    post_builtin_view_updates: Vec<StateUpdate>,
+    pre_builtin_view_additions: Vec<StateUpdate>,
+    builtin_view_additions: Vec<(&'static Builtin<NameReference>, GlobalId)>,
+    post_builtin_view_additions: Vec<StateUpdate>,
 }
 
+// TODO(jkosh44) When StateUpdates get a timestamp, this will have to return something like
+// `Vec<(SortedUpdates, Timestamp)>` that is sorted by timestamp.
 /// Sort [`StateUpdate`]s in dependency order.
-// TODO(jkosh44) It is very IMPORTANT that per timestamp, the updates are sorted retractions
-// then additions. Within the retractions the objects should be sorted in reverse dependency
-// order (objects->schema->database, replica->cluster, etc.). For now we panic on retractions.
 fn sort_updates(updates: Vec<StateUpdate>) -> SortedUpdates {
```

Comment:
The changes to this function were to add support for retractions. Builtin migrations need to retract old objects and add new ones.